pyo3-macros-backend/src/from_pyobject.rs

```diff
@@ -354,83 +324,72 @@ impl FieldAttribute {
         let metaitem = match list.nested.len() {
             0 => return Ok(None),
             1 => list.nested.into_iter().next().unwrap(),
-            _ => {
-                return Err(spanned_err(
-                    list.nested,
-                    "Only one of `item`, `attribute` can be provided, possibly with an \
-                     additional argument: `item(\"key\")` or `attribute(\"name\").",
-                ))
-            }
+            _ => bail_spanned!(
+                list.nested.span() =>
+                "only one of `attribute` or `item` can be provided"
+            ),
         };
         let meta = match metaitem {
             syn::NestedMeta::Meta(meta) => meta,
-            syn::NestedMeta::Lit(lit) => {
-                return Err(spanned_err(
-                    lit,
-                    "Expected `attribute` or `item`, not a literal.",
-                ))
-            }
+            syn::NestedMeta::Lit(lit) => bail_spanned!(
+                lit.span() =>
+                "expected `attribute` or `item`, got a literal"
+            ),
         };
         let path = meta.path();
         if path.is_ident("attribute") {
             Ok(Some(FieldAttribute::GetAttr(Self::attribute_arg(meta)?)))
         } else if path.is_ident("item") {
             Ok(Some(FieldAttribute::GetItem(Self::item_arg(meta)?)))
         } else {
-            Err(spanned_err(meta, "Expected `attribute` or `item`."))
+            bail_spanned!(meta.span() => "expected `attribute` or `item`");
         }
     }
 
     fn attribute_arg(meta: Meta) -> syn::Result<Option<syn::LitStr>> {
-        let arg_list = match meta {
+        let mut arg_list = match meta {
             Meta::List(list) => list,
             Meta::Path(_) => return Ok(None),
-            Meta::NameValue(nv) => {
-                let err_msg = "Expected a string literal or no argument: `pyo3(attribute(\"name\") or `pyo3(attribute)`";
-                return Err(spanned_err(nv, err_msg));
-            }
-        };
-        let arg_msg = "Expected a single string literal argument.";
-        let first = match arg_list.nested.len() {
-            1 => arg_list.nested.first().unwrap(),
-            _ => return Err(spanned_err(arg_list, arg_msg)),
+            Meta::NameValue(nv) => bail_spanned!(
+                nv.span() =>
+                "expected a string literal or no argument: `pyo3(attribute(\"name\")` or \
```

Comment:
Can't we avoid this newline?

Reply:
Without the newline the string goes to column 108, so in order to keep formatting consistent I used the newline 😢